deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/config/CompositeAuthenticationFactory.java

```diff
@@ -54,10 +52,7 @@ public ContainerRequestFilter create(TeletraanServiceContext context) throws Exc
                         .setAuthorizer(context.getAuthorizationFactory().create(context))
                         .buildAuthFilter();
 
-        List<AuthFilter> filters = new ArrayList<>();
-        filters.addAll(tokenFilters);
-        filters.add(envoyAuthFilter);
-
-        return new ChainedAuthFilter(filters);
+        return new ChainedAuthFilter(Arrays.asList(createScriptTokenAuthFilter(context),
+                createOauthTokenAuthFilter(context), envoyAuthFilter, createJwtTokenAuthFilter(context)));
```

Comment:
I don't disagree. `ServiceMeshAuthFilter` won't work because as you mentioned,Envoy can authenticate Oauth tokens via the HTTP proxy, which is different from the service mesh. Eventually we will only have Envoy. Realistically we will have only `ScriptToken`, `Jwt` and `Envoy` shortly.